### ISSUE TITLE
Externer Login für Authentifizierung

### ISSUE THREAD

**Issue by jzakotnik** [jzakotnik/openlibry#15]

**Comment by jornfranke** [comment]
Vorschlag:
1) [OpenAuth](https://en.wikipedia.org/wiki/OAuth) für Schulen, die keinen Identity Provider haben. Dann können sich die Schüler mit ihren Sozial-Networking Accounts einloggen
2) [SAML2](https://en.wikipedia.org/wiki/Security_Assertion_Markup_Language) und/oder [OIDC](https://en.wikipedia.org/wiki/OpenID) für Biblitotheken die schon eine eigene Infrastruktur für Identity haben.

Da schon next.js als Backend genutzt wird könnte man evt. [auth.js](https://authjs.dev/) einbinden. Es gibt sicherlich noch andere Bibliotheken und da sollte man mal einen Vergleich machen. Das kann zwar kein SAML2, aber viele Identity Provider unterstützen sowieso beides.

**Comment by jzakotnik** [comment]
Ja, OpenAuth wäre auch meine erste Wahl. Praktisch aber noch etwas unklar, da insbesondere bei kleinen Grundschulen die Ehrenamtlichen nicht unbedingt ihre privaten  Social Accounts nutzen wollen. Und ein Userverzeichnis gibts eh nicht. 

Alternative wäre noch einen eigenen Identity Provider im Container mit auszuliefern oder die Userverwaltung direkt in der DB zu machen (suboptimal, aber am einfachsten und wenigsten Maintenance)

**Comment by jzakotnik** [comment]
Braucht niemand
